### PR TITLE
Full text search improvements 

### PR DESCRIPTION
**Pull request by elcorbs**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/PA-410

### *What is the problem we're trying to solve*

There are a few improvements to make to the full text search after testing. 

### *What changes have we introduced*

- Added more synonyms for "road"
- Exclude number from the text search and then perform an exact match on the numbers in the address, if a number was in the search terms. This is because we don't want to fuzzily match building numbers

#### _Checklist_

- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: AddressesAPI/V2/Gateways/ElasticGateway.cs [hunk 1]
```diff
@@ -121,14 +121,21 @@ private static QueryContainer SearchAddressFullText(SearchParameters request, Qu
...
+            var fuzzyMatchText = q.Match(c => c
                 .Field(f => f.FullAddress)
                 .Query(request.AddressQuery)
                 .Analyzer("address_text")
                 .Fuzziness(Fuzziness.Auto)
-                .PrefixLength(3)
                 .Operator(Operator.And)
             );
+            var exactlyMatchNumbers = q.Match(m => m
+                .Field("full_address.extracted_numbers")
```
Comment (Renny4Real) [inline AddressesAPI/V2/Gateways/ElasticGateway.cs:132]: When indexing the addresses, do we create a separate field for numbers? Is that what `full_address.extracted_numbers` is referring to?

Comment (Renny4Real) [reply AddressesAPI/V2/Gateways/ElasticGateway.cs:132]: I see. It's created from the analyser in `data/elasticsearch/index.json`?

File: data/elasticsearch/index.json [hunk 2]
```diff
@@ -42,14 +47,27 @@
           "type": "synonym",
           "ignore_case": true,
           "lenient": false,
-          "synonyms": [ "road, lane, street, avenue, rd, str, avn, ave, close => road" ]
+          "synonyms": [ "road, raod, lane, lanes, street, streat, avenue, rd, str, avn, ave, close => road" ]
```
Comment (Renny4Real) [inline data/elasticsearch/index.json:50]: Interesting 🤔 , wouldn't `raod` and `streat` be treated as spelling mistakes for fuzzy matching to pick up? Why do we add them as synonyms here?

**Review by Renny4Real — APPROVED** [review]
🎉

File: AddressesAPI/V2/Gateways/ElasticGateway.cs (hunk 1 above)
Comment (elcorbs) [reply AddressesAPI/V2/Gateways/ElasticGateway.cs:132]: Yeah, it's a subfield. So it's the same data with different analyzers applied to it

File: data/elasticsearch/index.json (hunk 2 above)
Comment (elcorbs) [reply data/elasticsearch/index.json:50]: It doesn't fuzzily match when checking for synonyms unfortunately

**Pull request merged**